rules/experimental/rbe/README.md

```diff
@@ -0,0 +1,256 @@
+
+It is highly recommended to use a globally unique name for this repo (and definitely not
+"exec_properties") for reasons that are discussed in more details [below](#anti-patterns).
+
+## create_exec_properties_dict
+
+create_exec_properties_dict is a starlark macro that creates a dictionary of remote execution
+properties. create_exec_properties_dict ensures that the created dictionary is compatible with
+what RBE supports.
+
+It is highly recommended that this macro only be called from the WORKSPACE file in the context of
+creating repo rules using rbe_exec_properties or custom_exec_properties. See more on this below.
+
+## merge_dicts
+
+merge_dicts is a starlark macro that merges dictionaries of remote execution properties.
+
+## Use cases
+
+The following are some use cases showing how these repo rules should be used.
+
+### Use case 1 - The standard use case:
+
+In the WORKSPACE file, call
+```
+rbe_exec_properties(
+    name = "exec_properties",
+)
+```
+
+This creates a local repo @rbe_exec_properties with standard RBE execution property constants. For
+example, NETWORK_ON which is the dict {"dockerNetwork" : "standard"}. (For the full list of these
+constants see STANDARD_PROPERTY_SETS in exec_properties.bzl.)
+
+Then, in some BUILD file in your work space, you can reference this execution property constant as
+follows:
+```
+load("@exec_properties//:constants.bzl", "NETWORK_ON")
+...
+foo_test(
+   ...
+   exec_properties = NETWORK_ON,
+)
+```
+
+### Use case 2 - local execution
+
+If Bazel is set up so that the targets are executed locally, then the contents of exec_properties
+are ignored.
+
+### Use case 3 - non-RBE remote execution:
+
+Let's assume that the non-RBE remote execution endpoint provides a macro similar to
+rbe_exec_properties (say other_re_exec_properties), which populates the same constants (e.g.
+NETWORK_ON) with possibly different dict values.
+In this case, the WORKSPACE would, presumably, look like this:
+```
+other_re_exec_properties(
+    name = "exec_properties",
+)
+```
+
+And the targets in the BUILD files will be able to depend on targets from other repos that were
+written with RBE in mind, as the name of the repo defined in the WORKSPACE (exec_properties in this
+case) is the same. That is why the repo name exec_properties does *not* contain the word RBE.
+
+### Use case 4 - rbe_exec_properties with override_constants:
+
+Let's now assume that a particular repo, running with a particular RBE setup, wants to run
+all its remote execution actions without network access, possibly for the sake of identifying any
+network dependencies. This would be achieved as follows.
+
+In the WORKSPACE file, call
+```
+rbe_exec_properties(
+    name = "exec_properties",
+    override_constants = {
+        "NETWORK_ON": create_exec_properties_dict(docker_network = "off"),
+    },
+)
+```
+
+This would override the meaning of NETWORK_ON for this workspace only.
```

Comment:
capitalize WORKSPACE if referring to the file or use Bazel project (if that is what you are referring to) here and below

Reply:
done. Using "Bazel project" or "project" instead of "workspace" or "repo".